### ISSUE TITLE
BRexx370 Observations

### ISSUE THREAD

**Issue by NicClouston** [mvslovers/brexx370#74]
BRexx Comments

These comments result from my attempt to convert just one (a short one, at that) of my z/OS Rexx execs to run under BRexx370.

1 - PARSE SOURCE, in batch, does not return the name of the program being interpreted
2 - EXECIO
	. Return codes are not documented - need a minimum of all OK and EOF before reading all requested records plus diagnostic messages eg dataset not found, I/O error.
	. Gives a return code of 0 even though the stem to be written does not exist
	. Needs, as a minimum, the same functionality as that given by EXECIO from OORexx
		. DISKR
		. DISKRU
		. DISKW
		. OPEN
		. FINIS
		. specify number of records to read/write
3 - ADDRESS documentation does not list the environments. I used ADDRESS() to get and display the environment and it said SYSTEM but when I used ADDRESS SYSTEM to submit a batch job I got a return code of -1 as I did with the undocumented TSO. There is no documentation on what a -1 means.
4 - For documentation we are referred to that for the PC. Does that mean that ALL keywords/functions/options documented there are available in this version with that same synatx/option list? If not then differences should be explicitly noted. I would prefer a totally new documentation set - as a PDF (with the contents at the front and index at the back).

**Comment by mgrossmann** [comment]
Hello, Nic,

we are very happy that you are trying to use brexx/370. One thing you should always be aware of. brexx/370 is not IBM REXX. It is a nearly ANSI compatible REXX implementation. We try to implement some important TSO/E REXX features in our spare time. 

We will also be happy to look at some of your comments. If you find the format of the documentation unsatisfactory, or if there are things missing in the documentation, you are welcome to contribute to it. 

Have fun with brexx/370

**Comment by mgrossmann** [comment]
Hello, Nic,

As you have correctly noticed, there is no complete documentation for each of the features we have added. We try to describe each new function in a rudimentary way, but unfortunately, we don't have the capacity for documentation like IBM's. We apologize for this and invite you to help us with this. The original documentation is untouched by us. The largest part of this documentation should still exist. We have had to remove some things that cannot be used on the mainframe. A few functions have been extended. The original documentation together with the PDF's we deliver must be sufficient at this time.

Happy BREXXing
